presto-main/src/test/java/io/prestosql/server/TestDynamicFilterService.java

```diff
@@ -0,0 +1,368 @@
+import static org.testng.Assert.assertTrue;
+
+public class TestDynamicFilterService
+{
+    @Test
+    public void testDynamicFilterSummaryCompletion()
+    {
+        DynamicFilterService dynamicFilterService = new DynamicFilterService(new TaskManagerConfig());
+        String filterId = "df";
+        QueryId queryId = new QueryId("query");
+        StageId stageId = new StageId(queryId, 0);
+        List<TaskId> taskIds = ImmutableList.of(new TaskId(stageId, 0), new TaskId(stageId, 1), new TaskId(stageId, 2));
+
+        assertFalse(dynamicFilterService.getSummary(queryId, filterId).isPresent());
+        TestDynamicFiltersStageSupplier dynamicFiltersStageSupplier = new TestDynamicFiltersStageSupplier();
+        dynamicFiltersStageSupplier.addDynamicFilter(filterId, taskIds);
+        dynamicFilterService.registerQuery(queryId, dynamicFiltersStageSupplier);
+        assertFalse(dynamicFilterService.getSummary(queryId, filterId).isPresent());
+
+        dynamicFiltersStageSupplier.storeSummary(
+                filterId,
+                new TaskId(stageId, 0),
+                singleValue(INTEGER, 1L));
+        dynamicFilterService.collectDynamicFilters();
+        assertFalse(dynamicFilterService.getSummary(queryId, filterId).isPresent());
+
+        dynamicFiltersStageSupplier.storeSummary(
+                filterId,
+                new TaskId(stageId, 1),
+                singleValue(INTEGER, 2L));
+        dynamicFilterService.collectDynamicFilters();
+        assertFalse(dynamicFilterService.getSummary(queryId, filterId).isPresent());
+
+        dynamicFiltersStageSupplier.storeSummary(
+                filterId,
+                new TaskId(stageId, 2),
+                singleValue(INTEGER, 3L));
+        dynamicFilterService.collectDynamicFilters();
+        Optional<Domain> summary = dynamicFilterService.getSummary(queryId, filterId);
+        assertTrue(summary.isPresent());
+        assertEquals(summary.get(), multipleValues(INTEGER, ImmutableList.of(1L, 2L, 3L)));
+    }
+
+    @Test
+    public void testDynamicFilterSupplier()
+    {
+        DynamicFilterService dynamicFilterService = new DynamicFilterService(new TaskManagerConfig());
+        String filterId1 = "df1";
+        String filterId2 = "df2";
+        Expression df1 = expression("DF_SYMBOL1");
+        Expression df2 = expression("DF_SYMBOL2");
+        QueryId queryId = new QueryId("query");
+        StageId stageId1 = new StageId(queryId, 1);
+        StageId stageId2 = new StageId(queryId, 2);
+
+        Supplier<TupleDomain<ColumnHandle>> dynamicFilterSupplier = dynamicFilterService.createDynamicFilterSupplier(
+                queryId,
+                ImmutableList.of(
+                        new DynamicFilters.Descriptor(filterId1, df1),
+                        new DynamicFilters.Descriptor(filterId2, df2)),
+                ImmutableMap.of(
+                        Symbol.from(df1), new TestingColumnHandle("probeColumn"),
```

Comment:
add a dynamic filter here for a second column too which receives `none` domain

Reply:
Done